views/lib/components/files-storage/files-grid.hbs

```diff
@@ -0,0 +1,102 @@
+<section class="files">
+	<div class="row">
+		{{#each files}}
+			<div class="col-sm-12">
+				<div class="card file {{#ifeq this.securityCheck.status "blocked"}}btn-file-danger{{/ifeq}}" data-href="/files/file?file={{this.name}}" data-file-name="{{this.name}}" data-file-size="{{this.size}}" data-file-id="{{this.id}}">
+					<div class="openFile">
+						<div class="card-block
+							{{#ifneq this.securityCheck.status "blocked"}} fileviewer{{/ifneq}}"
+							{{#ifneq this.securityCheck.status "blocked"}} data-file-viewer-type="{{../this.type}}"{{/ifneq}}
+							{{#ifneq this.securityCheck.status "blocked"}} data-file-viewer-savename="{{../this.saveName}}"{{/ifneq}}
+							{{#ifneq this.securityCheck.status "blocked"}} data-file-viewer-id="{{../this.id}}"{{/ifneq}}
+							tabindex="0" aria-label="{{$t "files.label.clickToOpenFile" (dict "name" this.name)}}"
+							role="button">
+							<div class="card-title">
+								<div class="file-preview-wrapper col-sm-1 no-padding hidden-xs-down">
+									<div class="file-preview"
+										 style="background-image: url({{this.thumbnail}}); -webkit-background-size: 25px ;background-size: 25px;"></div>
+								</div>
+								<a class="col-sm-10 title">
+									{{this.name}}
+									{{#if this.size}}
+										<small style="float: right;" class="file-info" id="{{this.id}}"></small>
+									{{/if}}
+								</a>
+							</div>
+							<div class="card-text">
+							  <small class="file-location">{{#if this.context}}{{$t "files.text.foundIn"}} {{this.context}}{{/if}}</small><br>
+								{{#ifeq this.securityCheck.status "blocked"}}
+									<button
+									class="file-btn btn-file-danger"
+									title="{{$t "files.label.dangerousCode"}}">
+										<i aria-hidden="true" class="fa fa-exclamation-triangle"></i>
+									</button>
+								{{else}}
+									<button data-href="/api/v3/file/download/{{../this.id}}/{{../this.name}}"
+										class="file-btn"
+										data-method="download"
+										title="{{$t "files.label.downloadFile"}}">
+										<i aria-hidden="true" class="fa fa-cloud-download"></i>
+									</button>
+								{{/ifeq}}
+
+								<button
+								    data-file-id="{{this.id}}"
+									data-file-name="{{this.name}}"
+									data-file-path="{{this.path}}"
+									title="{{$t "files.label.renameFile"}}"
+									class="file-btn fa fa-edit file-name-edit">
+								</button>	
+								{{#userHasPermission "FILE_DELETE"}}
```

Comment:
If we use in backend FILE_STORAGE_* then we should also replace each of the outdated permissions.
We should also check if the vue client use the rights.